src/foundry/foundry.js/config.d.ts

```diff
@@ -1811,175 +1400,123 @@ declare global {
-
-        /**
-         * @defaultValue `LightingLayer`
-         */
-        lighting: ConstructorOf<LightingLayer>;
-
-        /**
-         * @defaultValue `SightLayer`
-         */
-        sight: ConstructorOf<SightLayer>;
-
-        /**
-         * @defaultValue `EffectsLayer`
-         */
-        effects: ConstructorOf<EffectsLayer>;
-
-        /**
-         * @defaultValue `ControlsLayer`
-         */
-        controls: ConstructorOf<ControlsLayer>;
-
-        [key: string]: ConstructorOf<CanvasLayer>;
+        /** @defaultValue `{ layerClass: BackgroundLayer, group: "primary" }` */
+        background: LayerDefinition<ConstructorOf<BackgroundLayer>>;
+
+        /** @defaultValue `{ layerClass: DrawingsLayer, group: "primary" }` */
+        drawings: LayerDefinition<ConstructorOf<DrawingsLayer>>;
+
+        /** @defaultValue `{ layerClass: GridLayer, group: "primary" }` */
+        grid: LayerDefinition<ConstructorOf<GridLayer>>;
+
+        /** @defaultValue `{ layerClass: TemplateLayer, group: "primary" }` */
+        templates: LayerDefinition<ConstructorOf<TemplateLayer>>;
+
+        /** @defaultValue `{ layerClass: TokenLayer, group: "primary" }` */
+        tokens: LayerDefinition<ConstructorOf<TokenLayer>>;
+
+        /** @defaultValue `{ layerClass: ForegroundLayer, group: "primary" }` */
+        foreground: LayerDefinition<ConstructorOf<ForegroundLayer>>;
+
+        /** @defaultValue `{ layerClass: WallsLayer, group: "effects" }` */
+        walls: LayerDefinition<ConstructorOf<WallsLayer>>;
+
+        /** @defaultValue `{ layerClass: LightingLayer, group: "effects" }` */
+        lighting: LayerDefinition<ConstructorOf<LightingLayer>>;
+
+        /** @defaultValue `{ layerClass: WeatherLayer, group: "effects" }` */
+        weather: LayerDefinition<ConstructorOf<CanvasLayer>>; // FIXME: WeatherLayer
+
+        /** @defaultValue `{ layerClass: SightLayer, group: "effects" }` */
+        sight: LayerDefinition<ConstructorOf<SightLayer>>;
+
+        /** @defaultValue `{ layerClass: SoundsLayer, group: "interface" }` */
+        sounds: LayerDefinition<ConstructorOf<SoundsLayer>>;
+
+        /** @defaultValue `{ layerClass: NotesLayer, group: "interface" }` */
+        notes: LayerDefinition<ConstructorOf<NotesLayer>>;
+
+        /** @defaultValue `{ layerClass: ControlsLayer, group: "interface" }` */
+        controls: LayerDefinition<ConstructorOf<ControlsLayer>>;
+
+        [key: string]: LayerDefinition<ConstructorOf<CanvasLayer>>;
+      }
+
+      // TODO: The type spec could be improved, if we add a types only interface common to the canvas groups. Otherwise
+      // the first common superclass they share is `PIXI.Container`.
+      interface GroupDefinition<GroupClass extends ConstructorOf<PIXI.Container>> {
+        groupClass: GroupClass;
+      }
+
+      interface LayerDefinition<LayerClass extends ConstructorOf<CanvasLayer>> {
+        layerClass: LayerClass;
+        group: keyof CONFIG['Canvas']['groups'];
```

Comment:
If we do it like this, we probably should make `groups` and interface, too, so that people can inject new groups on the type level via declaration merging.

Reply:
I will also do that in the canvas groups PR.